core/modules/util/IterableFormatter.h

```diff
@@ -0,0 +1,138 @@
+#include <vector>
+#include <iterator>
+#include <sstream>
+#include <string>
+
+// Third-party headers
+
+// LSST headers
+
+// Qserv headers
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+/**
+ *  Provide generic output operator for iterable data structures
+ *
+ *  Output format is [a, b, c, ...]
+ *  The elements stored in the iterable data structure
+ *  must provide an ouput operator.
+ *  Examples:
+ *  - works fine for std::vector<std::string>,
+ *  - doesn't work for std::multimap.
+ */
+template <typename Iter>
+class IterableFormatter
+{
+public:
+
+    /**
+     *  Create a printable wrapper for an iterable data structure
+     *
+     *  @param x:       an iterable data structure
+     *  @param first:   start printing from x[first]
+     *  @param open:    opening bracket, NULL is undefined behaviour
+     *  @param close:   closing bracket, NULL is undefined behaviour
+     *  @param sep:     separator between elements, NULL is undefined behaviour
+     */
+    explicit IterableFormatter(Iter begin, Iter end,
+                               char const *const open, char const *const close,
+                               char const *const sep) :
+            _begin(begin), _end(end), _open(open), _close(close), _sep(sep) {
+    }
+
+    /**
+     *  Output operator for IterableFormatter<Iterable>
+     *
+     *  @param out
+     *  @param iterableFormatter
+     *  @return an output stream, with no newline at the end
+     */
+    friend std::ostream& operator<<(std::ostream& os,
+                                    const IterableFormatter<Iter>& self) {
+        os << self._open;
+        auto last =  std::prev(self._end);
+        for (auto it = self._begin; it != last; ++it) {
+            os << *it << self._sep << " ";
+        }
+        if (last != self._end) {
+            os << *last;
+        }
+        os << self._close;
+        return os;
+    }
+
+private:
+    Iter _begin;
+    Iter _end;
+    char const *const _open;
+    char const *const _close;
+    char const *const _sep;
+};
+
+template <typename Iterator>
+IterableFormatter<Iterator> printable_iter(Iterator begin, Iterator end,
+                                  char const *const open = "[", char const *const close = "]",
+                                  char const *const sep = ",")
+{
+    return IterableFormatter<Iterator>(begin, end, open, close, sep);
+}
+
+
+/**
+ *  Create a printable wrapper for an iterable data structure
+ *
+ *  @param x:       an iterable data structure
+ *  @param first:   start printing from x[first]
+ *  @param open:    opening bracket, NULL is undefined behaviour
+ *  @param close:   closing bracket, NULL is undefined behaviour
+ *  @param sep:     separator between elements, NULL is undefined behaviour
+ *  @return:        an object wrapping x and providing an output operator
+ */
+template <typename Iterable>
+IterableFormatter<typename Iterable::const_iterator> printable(Iterable const& x, int first = 0,
```

Comment:
My idea was to remove `first` from here, if you want to print part of the collection then use iterators.